sys/psa_crypto/psa_crypto_algorithm_dispatch.c

```diff
@@ -554,32 +554,63 @@ psa_status_t psa_algorithm_dispatch_cipher_encrypt( const psa_key_attributes_t *
                                                     size_t output_size,
                                                     size_t *output_length)
 {
-    psa_cipher_op_t op = PSA_ENCODE_CIPHER_OPERATION(alg, attributes->bits, attributes->type);
     uint8_t *key_data = NULL;
     size_t *key_bytes = NULL;
 
     psa_get_key_data_from_key_slot(slot, &key_data, &key_bytes);
-
-    if (op == PSA_INVALID_OPERATION) {
-        return PSA_ERROR_INVALID_ARGUMENT;
-    }
-
-    switch (op) {
-#if IS_USED(MODULE_PSA_CIPHER_AES_128_CBC)
-    case PSA_CBC_NO_PAD_AES_128:
-        return psa_cipher_cbc_aes_128_encrypt(attributes, key_data, *key_bytes, alg, input,
-                                              input_length, output, output_size, output_length);
-#endif
-#if IS_USED(MODULE_PSA_CIPHER_AES_192_CBC)
-    case PSA_CBC_NO_PAD_AES_192:
-        return psa_cipher_cbc_aes_192_encrypt(attributes, key_data, *key_bytes, alg, input,
-                                              input_length, output, output_size, output_length);
-#endif
-#if IS_USED(MODULE_PSA_CIPHER_AES_256_CBC)
-    case PSA_CBC_NO_PAD_AES_256:
-        return psa_cipher_cbc_aes_256_encrypt(attributes, key_data, *key_bytes, alg, input,
-                                              input_length, output, output_size, output_length);
-#endif
+    
+    switch (alg) {
+    #if IS_USED(MODULE_PSA_CIPHER_AES_128_CBC)||\
+        IS_USED(MODULE_PSA_CIPHER_AES_192_CBC)||\
+        IS_USED(MODULE_PSA_CIPHER_AES_256_CBC)||\
+        IS_USED(MODULE_PSA_CIPHER_CHACHA_20)
+    case PSA_ALG_CBC_NO_PADDING:
+        if (attributes->type != PSA_KEY_TYPE_AES) {
+            return PSA_ERROR_INVALID_ARGUMENT;
+        }
+        switch (attributes->bits) {
```

Comment:
nested switch-statements are not that readable, why did you decide to remove the macro-based approach and replace it with that?